activation/nipost.go

```diff
@@ -388,23 +366,74 @@ func (nb *NIPostBuilder) submitPoetChallenge(
 	})
 }
 
-// Submit the challenge to all registered PoETs.
+// Submit the challenge to registered PoETs with given addresses.
 func (nb *NIPostBuilder) submitPoetChallenges(
 	ctx context.Context,
 	signer *signing.EdSigner,
-	deadline time.Time,
+	poetProofDeadline time.Time,
+	curPoetRoundStartDeadline time.Time,
 	challenge []byte,
-) error {
+) ([]nipost.PoETRegistration, error) {
+	// check if some registrations missing
+	existingRegistrations, err := nipost.PoetRegistrationsByNodeId(nb.localDB, signer.NodeID())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get poet registrations from db: %w", err)
+	}
+
+	var i int
+	addresses := make([]string, len(nb.poetProvers))
+	for _, client := range nb.poetProvers {
+		addresses[i] = client.Address()
+		i++
+	}
+
+	existingRegistrationsMap := make(map[string]struct{})
+	for _, er := range existingRegistrations {
+		existingRegistrationsMap[er.Address] = struct{}{}
+	}
+
+	missingRegistrationsPoETAddresses := make(map[string]struct{})
+	for _, addr := range addresses {
+		if _, ok := existingRegistrationsMap[addr]; !ok {
+			missingRegistrationsPoETAddresses[addr] = struct{}{}
+		}
+	}
+
+	if len(missingRegistrationsPoETAddresses) == 0 {
+		return existingRegistrations, nil
+	}
```

Comment:
I think it should not return the poets removed from the config here (but it shouldn't remove their registrations from the DB yet - in case user changes his mind and adds them back to the config).

The use case is, for example, the user used a custom poet service that died/was removed. There is no point in later querying this poet for the proof.

Reply:
We return here only already existing registrations and only in case if no registrations missing. 
So it means, that user restarted node, but changed nothing in PoETs config and it makes no sense to make further checks